game/world/managers/objects/units/creature/CreatureManager.py

```diff
@@ -77,35 +77,70 @@ def __init__(self, **kwargs):
         self.has_parry_passive = True
 
     # This can also be used to 'morph' the creature.
-    def initialize_from_creature_template(self, creature_template, subtype=CustomCodes.CreatureSubtype.SUBTYPE_GENERIC):
+    def initialize_from_creature_template(
+        self, 
+        creature_template, 
+        subtype=CustomCodes.CreatureSubtype.SUBTYPE_GENERIC):
+
         if not creature_template:
             return
-
+            
         self.entry = creature_template.entry
         self.creature_template = creature_template
         self.entry = self.creature_template.entry
         self.class_ = self.creature_template.unit_class
-        self.resistance_0 = self.creature_template.armor
-        self.resistance_1 = self.creature_template.holy_res
-        self.resistance_2 = self.creature_template.fire_res
-        self.resistance_3 = self.creature_template.nature_res
-        self.resistance_4 = self.creature_template.frost_res
-        self.resistance_5 = self.creature_template.shadow_res
         self.npc_flags = self.creature_template.npc_flags
         self.static_flags = self.creature_template.static_flags
         self.regen_flags = self.creature_template.regeneration
         self.virtual_item_info = {}  # Slot: VirtualItemInfoHolder
         self.base_attack_time = self.creature_template.base_attack_time
         self.ranged_attack_time = self.creature_template.ranged_attack_time
-        self.ranged_dmg_min = self.creature_template.ranged_dmg_min
-        self.ranged_dmg_max = self.creature_template.ranged_dmg_max
         self.unit_flags = self.creature_template.unit_flags
         self.emote_state = 0
         self.faction = self.creature_template.faction
         self.creature_type = self.creature_template.type
         self.spell_list_id = self.creature_template.spell_list_id
         self.sheath_state = WeaponMode.NORMALMODE
         self.subtype = subtype
+        self.level = randint(self.creature_template.level_min, self.creature_template.level_max)
+        
+        # Specific stats for class and level
+        creature_class_level_stats = self.get_creature_class_level_stats()
+
+        melee_dmg_min, melee_dmg_max = CreatureFormulas.calculate_min_max_damage(
+            creature_class_level_stats.melee_damage,
+            creature_template.damage_multiplier,
+            creature_template.damage_variance
+        )
+
+        ranged_dmg_min, ranged_dmg_max = CreatureFormulas.calculate_min_max_damage(
+            creature_class_level_stats.ranged_damage,
+            creature_template.damage_multiplier,
+            creature_template.damage_variance
+        )
+
+        # Stats
+        self.dmg_min = melee_dmg_min
+        self.dmg_max = melee_dmg_max
+        self.ranged_dmg_min = ranged_dmg_min
+        self.ranged_dmg_max = ranged_dmg_max
+        self.resistance_0 = int(creature_class_level_stats.armor * creature_template.armor_multiplier)
```

Comment:
agility should be added to the total armor value 1 to 1

Reply:
![image](https://github.com/The-Alpha-Project/alpha-core/assets/8654362/29847f58-d315-46a2-9b93-1d56a73eed2d)
Seems like agility didn't increase armor in 0.5.3, at least for players. :thinking:

Reply:
Do we have any Beast Lore screenshots from alpha? We could see if armor includes agility or not from that.

Reply:
Don't think so, Beast Lore didn't even exist in 0.5.3.